MaterialLibrary/src/main/java/io/doist/material/elevation/ElevationDelegate.java

```diff
@@ -0,0 +1,360 @@
+        Drawable background = mView.getBackground();
+        if (background instanceof ElevationWrapperDrawable) {
+            background = ((ElevationWrapperDrawable) background).getWrappedDrawable();
+        }
+        return background;
+    }
+
+    /**
+     * Returns the {@link View}'s left padding excluding the extra added for the shadow.
+     */
+    public int getOriginalPaddingLeft() {
+        int paddingLeft = mView.getPaddingLeft();
+        ElevationWrapperDrawable elevationDrawable = getElevationDrawableWrapper();
+        if (elevationDrawable != null) {
+            paddingLeft -= elevationDrawable.getPaddingLeft();
+        }
+        return paddingLeft;
+    }
+
+    /**
+     * Returns the {@link View}'s top padding excluding the extra added for the shadow.
+     */
+    public int getOriginalPaddingTop() {
+        int paddingTop = mView.getPaddingTop();
+        ElevationWrapperDrawable elevationDrawable = getElevationDrawableWrapper();
+        if (elevationDrawable != null) {
+            paddingTop -= elevationDrawable.getPaddingTop();
+        }
+        return paddingTop;
+    }
+
+    /**
+     * Returns the {@link View}'s right padding excluding the extra added for the shadow.
+     */
+    public int getOriginalPaddingRight() {
+        int paddingRight = mView.getPaddingRight();
+        ElevationWrapperDrawable elevationDrawable = getElevationDrawableWrapper();
+        if (elevationDrawable != null) {
+            paddingRight -= elevationDrawable.getPaddingRight();
+        }
+        return paddingRight;
+    }
+
+    /**
+     * Returns the {@link View}'s bottom padding excluding the extra added for the shadow.
+     */
+    public int getOriginalPaddingBottom() {
+        int paddingBottom = mView.getPaddingBottom();
+        ElevationWrapperDrawable elevationDrawable = getElevationDrawableWrapper();
+        if (elevationDrawable != null) {
+            paddingBottom -= elevationDrawable.getPaddingBottom();
+        }
+        return paddingBottom;
+    }
+
+    /**
+     * Returns the {@link View}'s start padding excluding the extra added for the shadow.
+     */
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+    public int getOriginalPaddingStart() {
+        return mView.getLayoutDirection() == View.LAYOUT_DIRECTION_RTL ? getOriginalPaddingRight()
+                                                                       : getOriginalPaddingLeft();
+    }
+
+    /**
+     * Returns the {@link View}'s end padding excluding the extra added for the shadow.
+     */
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+    public int getOriginalPaddingEnd() {
+        return mView.getLayoutDirection() == View.LAYOUT_DIRECTION_RTL ? getOriginalPaddingLeft()
+                                                                       : getOriginalPaddingRight();
+    }
+
+    /**
+     * Returns a copy of the {@link View}'s {@link ViewGroup.LayoutParams} excluding the size / margin modifications.
```

Comment:
I was going to fix the docs, but instead fixed the method. It should provide a consistent API, ie. always return a copy or never return a copy. Since modifying the real layout params is not a good idea, it now returns a copy at all times.